libwallet/internal/politeia/politeia_client.go

```diff
@@ -132,25 +132,27 @@ func (c *politeiaClient) batchProposals(tokens []string) ([]Proposal, error) {
 }
 
 func getProposalType(payload string) ProposalType {
+	prop := ProposalTypeNormal
 	decodedBytes, err := base64.StdEncoding.DecodeString(payload)
 	if err != nil {
-		fmt.Println("error decoding proposalmetadata:", err)
-		return ProposalTypeNormal
+		fmt.Println("error decoding proposal metadata:", err)
+		return prop
 	}
 
 	var meta metadataProposal
 	err = json.Unmarshal(decodedBytes, &meta)
-	if err != nil {
+	if err == nil {
 		log.Error("error unmarshalling metadataProposal:", err)
+		return prop
```

Comment:
why are we checking if error == nil and then trying to log an error when it's nil

Reply:
Dang! `err != nil`